bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -362,10 +362,37 @@ impl Connection {
     pub fn negotiate(&mut self) -> Poll<Result<&mut Self, Error>> {
         let mut blocked = s2n_blocked_status::NOT_BLOCKED;
 
-        match unsafe { s2n_negotiate(self.connection.as_ptr(), &mut blocked).into_result() } {
-            Ok(_) => Ok(self).into(),
-            Err(err) if err.is_retryable() => Poll::Pending,
-            Err(err) => Err(err).into(),
+        unsafe {
+            s2n_negotiate(self.connection.as_ptr(), &mut blocked)
+                .into_poll()
+                .map_ok(|_| self)
+        }
+    }
+
+    pub fn send(&mut self, buf: &[u8]) -> Poll<Result<usize, Error>> {
+        let mut blocked = s2n_blocked_status::NOT_BLOCKED;
+        let buf_len: isize = buf.len().try_into().map_err(|_| Error::InvalidInput)?;
+        let buf_ptr = buf.as_ptr() as *const ::libc::c_void;
```

Comment:
[s2n_send](https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L458-L459) probably takes a void* because [posix send](https://man7.org/linux/man-pages/man2/send.2.html) takes a void*. But our rust bindings don't have to continue the tradition!